### PR TITLE
syncer: do not try to handle jobs anymore if the context is done

### LINKED ISSUE #1254: `pause-task` returns some connections errors
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

`pause-task`

2. What did you expect to see?

the task paused without errors.

3. What did you see instead?

```
"ErrCode": 10006,
"ErrClass": "database",
"ErrScope": "not-set",
"ErrLevel": "high",
"Message": "startLocation: [position: (mysql-bin.000029, 971322922), gtid-set: ], endLocation: [position: (mysql-bin.000029, 971323148), gtid-set: ]: execute statement failed: begin",
"RawCause": "sql: connection is already closed",
"Workaround": ""
```

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        Release Version: v2.0.0-beta.2-197-g5003e431
        Git Commit Hash: 5003e431f07169377f9afa3739ed0dace54be2fb
        Git Branch: master
        UTC Build Time: 2020-10-23 11:07:11
        Go Version: go version go1.13 linux/amd64
        ```
[pause-task-connection.log](https://github.com/pingcap/dm/files/5464427/pause-task-connection.log)

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

fix #1254

### What is changed and how it works?

do not try to handle jobs anymore if the context is done

### Check List 

Tests 

 - Unit test
 - Integration test

**NOTE**: this case is hard to test in CI now, `incremental_mode` may have a little effect on this.

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -1039,6 +1039,20 @@ func (s *Syncer) sync(tctx *tcontext.Context, queueBucket string, db *DBConn, jo
...
+			select {
+			case <-tctx.Ctx.Done():
+				// do not handle jobs anymore, because it should be failed with a done context.
+				// and avoid some errors like:
+				//  - `driver: bad connection` for `BEGIN`
+				//  - `sql: connection is already closed` for `BEGIN`
+				tctx.L().Debug("skip a remaining DML job in the job chan", zap.Stringer("job", sqlJob), zap.Int("remain", len(jobChan)))
+				s.execError.Set(tctx.Ctx.Err()) // set exexError to avoid flushing checkpoint because some jobs are skipped.
+				s.jobWg.Done()                  // done WaitGroup to avoid blocking
+				continue
```
Comment (lance6716) [inline syncer/syncer.go:1052]: how about `break` here or use `fatalF`. `continue` could consume `jobChan` and return but code still might go to `case <-time.After(waitTime)` and execute already saved `jobs`

Comment (csuzhangxc) [reply syncer/syncer.go]: move `tctx.Ctx.Done()` check into `executeSQLs` in https://github.com/pingcap/dm/pull/1295/commits/86a09c1b6e583a8e87c40355e6d24649aaeb8464.

**Comment by lance6716** [comment]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**